source/PlayerInfoPanel.cpp

```diff
@@ -655,13 +767,72 @@ void PlayerInfoPanel::DrawFleet(const Rectangle &bounds)
 
 
 
+// Sorts the player's fleet given a comparator function (based on column).
+void PlayerInfoPanel::SortShips(InfoPanelState::ShipComparator &shipComparator)
+{
+	// Save selected ships to preserve selection after sort.
+	set<shared_ptr<Ship>> selectedShips;
+	shared_ptr<Ship> lastSelected = panelState.SelectedIndex() == -1
+	? nullptr
+	: panelState.Ships()[panelState.SelectedIndex()];
+	
+	for(int i : panelState.AllSelected())
+		selectedShips.insert(panelState.Ships()[i]);
+	panelState.AllSelected().clear();
+	
+	// Move flagship to first position
+	for(auto &ship : panelState.Ships())
+		if(ship.get() == player.Flagship())
+		{
+			iter_swap(&ship, panelState.Ships().begin());
```

Comment:
How portable is using a pointer as an iterator here? E.g. will this fail if we change `panelState.Ship()` to not be a vector? Or if some standard library implements std::vector differently?

Reply:
changed to std::swap